src/stransport_stream.c

```diff
@@ -114,14 +114,20 @@ static OSStatus write_cb(SSLConnectionRef conn, const void *data, size_t *len)
 {
 	git_stream *io = (git_stream *) conn;
 	ssize_t ret;
+	size_t left, requested;
 
-	ret = git_stream_write(io, data, *len, 0);
-	if (ret < 0) {
-		*len = 0;
-		return -1;
-	}
+	requested = left = *len;
+	do {
+		ret = git_stream_write(io, data + (requested - left), left, 0);
+		if (ret <= 0) { /* There was an error or nothing could be written because stream is closed */
+			*len = 0;
+			return errSSLClosedAbort;
+		}
+
+		left -= ret;
+	} while (left);
 
-	*len = ret;
+	*len = requested;
```

Comment:
This change should not be necessary. The streams return when there is an error or when everything is written.